src/neorados/RADOS.cc

```diff
@@ -791,6 +791,10 @@ RADOS& RADOS::operator =(RADOS&&) = default;
 
 RADOS::~RADOS() = default;
 
+void RADOS::dup_from_librados(librados::Rados& rados) {
```

Comment:
I'm getting this error when trying the above:
```
/usr/bin/ld: lib/librbd_internal.a(NativeFormat.cc.o): in function `librbd::migration::NativeFormat<librbd::ImageCtx>::open(Context*)':
/home/oro/ceph/src/librbd/migration/NativeFormat.cc:235: undefined reference to `neorados::RADOS::operator=(neorados::RADOS&&)'
collect2: error: ld returned 1 exit status
```

Reply:
Weird.  Had libneorados not been linked in, you should have gotten a similar error for `dup_from_librados()` use in `NativeFormat.cc` because both are defined in the same file (just 4 lines apart):
```
 790 RADOS& RADOS::operator =(RADOS&&) = default;
 791 
 792 RADOS::~RADOS() = default;
 793 
 794 void RADOS::dup_from_librados(librados::Rados& rados) {
 795   impl.reset(new detail::RadosClient(rados.client));
 796 }
```

Reply:
I tried the following locally and it works for me:
```
-  m_image_ctx->rados_api.dup_from_librados(*rados_ptr);
+  m_image_ctx->rados_api = neorados::RADOS::make_with_librados(*rados_ptr);
```
Perhaps try rebuilding everything from scratch?  Otherwise you would probably need to dig into `src/librbd/CMakeLists.txt`.

Reply:
Ah, you didn't mention that it was failing to build/link tests ;)